pkg/kubectl/cmd/create_service.go

```diff
@@ -0,0 +1,217 @@
+		}
+	default:
+		return cmdutil.UsageError(cmd, fmt.Sprintf("Generator: %s not supported.", generatorName))
+	}
+	return RunCreateSubcommand(f, cmd, cmdOut, &CreateSubcommandOptions{
+		Name:                name,
+		StructuredGenerator: generator,
+		DryRun:              cmdutil.GetDryRunFlag(cmd),
+		OutputFormat:        cmdutil.GetFlagString(cmd, "output"),
+	})
+}
+
+const (
+	serviceNodePortLong = `
+Create a nodeport service with the specified name.`
+
+	serviceNodePortExample = `  # Create a new nodeport service named my-ns
+  kubectl create service nodeport my-ns --tcp=5678:8080`
+)
+
+// NewCmdCreateServiceNodePort is a macro command for creating secrets to work with Docker registries
+func NewCmdCreateServiceNodePort(f *cmdutil.Factory, cmdOut io.Writer) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "nodeport NAME [--tcp=port:targetPort] [--dry-run]",
+		Short:   "Create a NodePort service.",
+		Long:    serviceNodePortLong,
+		Example: serviceNodePortExample,
+		Run: func(cmd *cobra.Command, args []string) {
+			err := CreateServiceNodePort(f, cmdOut, cmd, args)
+			cmdutil.CheckErr(err)
+		},
+	}
+	cmdutil.AddApplyAnnotationFlags(cmd)
+	cmdutil.AddValidateFlags(cmd)
+	cmdutil.AddPrinterFlags(cmd)
+	cmdutil.AddGeneratorFlags(cmd, cmdutil.ServiceNodePortGeneratorV1Name)
+	addPortFlags(cmd)
+	return cmd
+}
+
+// CreateServiceNodePort is the implementation of the create secret docker-registry command
+func CreateServiceNodePort(f *cmdutil.Factory, cmdOut io.Writer, cmd *cobra.Command, args []string) error {
+	name, err := NameFromCommandArgs(cmd, args)
+	if err != nil {
+		return err
+	}
+	var generator kubectl.StructuredGenerator
+	switch generatorName := cmdutil.GetFlagString(cmd, "generator"); generatorName {
+	case cmdutil.ServiceNodePortGeneratorV1Name:
+		generator = &kubectl.ServiceCommonGeneratorV1{
+			Name:      name,
+			TCP:       cmdutil.GetFlagStringSlice(cmd, "tcp"),
+			Type:      api.ServiceTypeNodePort,
+			ClusterIP: "",
+		}
+	default:
+		return cmdutil.UsageError(cmd, fmt.Sprintf("Generator: %s not supported.", generatorName))
+	}
+	return RunCreateSubcommand(f, cmd, cmdOut, &CreateSubcommandOptions{
+		Name:                name,
+		StructuredGenerator: generator,
+		DryRun:              cmdutil.GetDryRunFlag(cmd),
+		OutputFormat:        cmdutil.GetFlagString(cmd, "output"),
+	})
+}
+
+const (
+	serviceLoadBalancerLong = `
+Create a LoadBalancer service with the specified name.`
+
+	serviceLoadBalancerExample = `  # Create a new nodeport service named my-lbs
+  kubectl create service loadbalancer my-lbs --tcp=5678:8080 --ingressIP=10.0.4.5`
+)
+
+// NewCmdCreateServiceLoadBalancer is a macro command for creating secrets to work with Docker registries
+func NewCmdCreateServiceLoadBalancer(f *cmdutil.Factory, cmdOut io.Writer) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "loadbalancer NAME [--tcp=port:targetPort] [--ingressIP=ipaddress] [--dry-run]",
```

Comment:
Where are you declaring/using `ingressIP`? Also use either `--ingress-ip` or `--ingressip`.